src/pages/ParticipantsPage.js

```diff
@@ -0,0 +1,141 @@
+import React from 'react';
+import {
+    View,
+} from 'react-native';
+import PropTypes from 'prop-types';
+import {withOnyx} from 'react-native-onyx';
+import styles from '../styles/styles';
+import ONYXKEYS from '../ONYXKEYS';
+import HeaderWithCloseButton from '../components/HeaderWithCloseButton';
+import Navigation from '../libs/Navigation/Navigation';
+import ScreenWrapper from '../components/ScreenWrapper';
+import OptionsList from '../components/OptionsList';
+import ROUTES from '../ROUTES';
+import CONST from '../CONST';
+
+const propTypes = {
+    /* Onyx Props */
+    // The personal details of the person who is logged in
+    myPersonalDetails: PropTypes.shape({
+        // Display name of the current user from their personal details
+        displayName: PropTypes.string,
+
+        // Avatar URL of the current user from their personal details
+        avatar: PropTypes.string,
+
+        // Login of the current user from their personal details
+        login: PropTypes.string,
+    }).isRequired,
+
+    // List of reports
+    reports: PropTypes.shape({
+        // The list of icons
+        icons: PropTypes.arrayOf(PropTypes.string),
+
+        // The report name
+        reportName: PropTypes.string,
+
+        // Array of participants
+        participants: PropTypes.arrayOf(PropTypes.string),
+    }).isRequired,
+
+    // Route params
+    route: PropTypes.shape({
+        params: PropTypes.shape({
+            // Report ID passed via route /participants/:reportID
+            reportID: PropTypes.string,
+        }),
+    }).isRequired,
+
+    // The chat priority mode
+    priorityMode: PropTypes.string.isRequired,
+};
+
+/**
+ * Returns all the participants in the active report
+ *
+ * @param {Object} report The active report object
+ * @param {Object} personalDetail The personal detail of the logged user
+ * @return {Array}
+ */
+const getAllParticipants = (report, personalDetail) => {
+    const {icons, participants, reportName} = report;
+    const displayNames = reportName.split(', ');
+
+    const members = participants.reduce((list, login, idx) => {
+        list.push({
+            alternateText: login,
+            displayName: displayNames[idx],
```

Comment:
This works, but it's not really the correct way to access these display names and will break in a future where report names are not structured as they are today.

Instead we should subscribe to the `personalDetails` key. Then grab everything we need like this:

```js
const userPersonalDetails = props.personalDetails[login];
const displayName = userPersonalDetails.displayName;
const avatar = userPersonalDetails.avatar;
```

etc.

Reply:
Thank you for the suggestion, will make the change.